src/common/lib/client/realtimechannel.ts

```diff
@@ -1016,6 +1016,23 @@ class RealtimeChannel extends EventEmitter {
   async status(): Promise<API.ChannelDetails> {
     return this.client.rest.channelMixin.status(this);
   }
+
+  async ensureAttached(): Promise<void> {
```

Comment:
Could we have a documentation comment here, please? It's quite a generic name but it's intended for a pretty specific scenario (especially once we align this further with the spec).